torchao/dtypes/nf4tensor.py

```diff
@@ -37,11 +41,244 @@ def decorator(func):
+            outer_wrapper.stride(),
+            outer_wrapper.storage_offset(),
+            outer_wrapper.dtype,
+            outer_wrapper.device,
+            outer_wrapper.requires_grad,
+        ),
+        outer_wrapper.block_size,
+        outer_wrapper.n_blocks,
+        outer_wrapper.scaler_block_size,
+        quantized_scalers_new_zeros,
+        quantization_factor_new_zeros,
+        outer_wrapper.scaler_mean,
+        quantized_data_new_zeros,
+        outer_wrapper.nf4,
+    )
+
+@implements(
+    [
+        aten.slice.Tensor,
+    ]
+)
+def nf4_slice(aten_op, args, kwargs=None):
+    if len(args) == 5:
+        raise NotImplementedError(f"aten.slice(NF4Tensor) with step={args[4]}")
+    if not args[1] == 0:
+        raise NotImplementedError(f"aten.slice(NF4Tensor) with dim={args[1]}")
+    if not args[2] == 0:
+        raise NotImplementedError(f"aten.slice(NF4Tensor) with start={args[2]}")
+    # for tensor 512 x 512, tensor[:, :512] dispatch to
+    # aten.slice(dim = 0, end=sys.maxsize)
+    if not args[3] in [args[0].size(0), sys.maxsize]:
+        raise NotImplementedError(f"aten.slice(NF4Tensor) with end={args[3]}")
+    return NF4Tensor(
+        SubclassTensorArgs(
+            args[0].size(),
+            args[0].stride(),
+            args[0].storage_offset(),
+            args[0].dtype,
+            args[0].device,
+            args[0].requires_grad,
+        ),
+        args[0].block_size,
+        args[0].n_blocks,
+        args[0].scaler_block_size,
+        args[0].quantized_scalers,
+        args[0].quantization_factor,
+        args[0].scaler_mean,
+        args[0].quantized_data,
+        args[0].nf4,
+    )
+
+@implements(
+    [
+        aten.view.default,
+    ]
+)
+def nf4_view(aten_op, args, kwargs=None):
+    size = args[1]
+    if not (len(size) == 1 and size[0] == -1):
+        raise NotImplementedError(f"aten.view(NF4Tensor) with size {size}")
+    quantized_scalers = aten_op(args[0].quantized_scalers, *(args[1:]), **kwargs)
+    quantization_factor = aten_op(args[0].quantization_factor, *(args[1:]), **kwargs)
+    quantized_data = aten_op(args[0].quantized_data, *(args[1:]), **kwargs)
+    tensor_meta = SubclassTensorArgs(
+        [args[0].numel()],
+        (1, ),
+        args[0].storage_offset(),
+        args[0].dtype,
+        args[0].device,
+        args[0].requires_grad,
+    )
+    return NF4Tensor(
+        tensor_meta,
+        args[0].block_size,
+        args[0].n_blocks,
+        args[0].scaler_block_size,
+        quantized_scalers,
+        quantization_factor,
+        args[0].scaler_mean,
+        quantized_data,
+        args[0].nf4,
+    )
+
+@implements(
+    [
+        aten.as_strided.default,
+    ]
+)
+def nf4_as_strided(aten_op, args, kwargs=None):
+    size = args[1]
+    stride = tuple(args[2])
+    storage_offset = args[3]
+    if not len(size) <= 2:
+        raise NotImplementedError(f"aten.as_strided(NF4Tensor) only support dim <= 2 but got dim={len(size)}")
```

Comment:
Seems like the above snippet might be useful since this seems to come up a lot. You could turn it into a generic decorator for input validation.